pkg/tls/rootca/rootca.go

```diff
@@ -152,14 +151,9 @@ func (w *watcher) loadRootCAsFile() (bool, RootCAs, error) {
 
 	w.log.Info("updating root CAs from file")
 
-	rootCAsCerts, err := pki.DecodeX509CertificateChainBytes(rootCAsPEM)
-	if err != nil {
-		return false, RootCAs{}, fmt.Errorf("failed to decode root CAs in certificate file %q: %w", w.filepath, err)
-	}
-
 	rootCAsPool := x509.NewCertPool()
-	for _, rootCert := range rootCAsCerts {
-		rootCAsPool.AddCert(rootCert)
+	if !rootCAsPool.AppendCertsFromPEM(rootCAsPEM) {
+		return false, RootCAs{}, fmt.Errorf("failed to decode root CAs in certificate file %q", w.filepath)
```

Comment:
`AppendCertsFromPEM` uses `encoding/pem` [under the hood](https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/crypto/x509/cert_pool.go;l=222) which is the vulnerable package we worked around when fixing https://github.com/cert-manager/cert-manager/security/advisories/GHSA-r4pg-vg54-wxx4

I don't think we need to rush to remove all uses of it, but I think there's something to be said in avoiding new uses of it if we can?

Reply:
Ok. Agreed.